sql3/planner/inbuiltfunctionsdate.go

```diff
@@ -466,6 +488,76 @@ func (n *callPlanExpression) EvaluateDatetimeAdd(currentRow []interface{}) (inte
 		return nil, sql3.NewErrCallParameterValueInvalid(0, 0, timeunit, "timeunit")
 	}
 }
+func (n *callPlanExpression) EvaluateDateTrunc(currentRow []interface{}) (interface{}, error) {
+	intervalEval, err := n.args[0].Evaluate(currentRow)
+	if err != nil {
+		return nil, err
+	}
+
+	dateEval, err := n.args[1].Evaluate(currentRow)
+	if err != nil {
+		return nil, err
+	}
+
+	// nil if anything is nil
+	if intervalEval == nil || dateEval == nil {
+		return nil, nil
+	}
+
+	//get the date value
+	coercedDate, err := coerceValue(n.args[1].Type(), parser.NewDataTypeTimestamp(), dateEval, parser.Pos{Line: 0, Column: 0})
+	if err != nil {
+		return nil, err
+	}
+
+	date, dateOk := coercedDate.(time.Time)
+	if !dateOk {
+		return nil, sql3.NewErrInternalf("unable to convert value")
+	}
+
+	//get the interval value
+	coercedInterval, err := coerceValue(n.args[0].Type(), parser.NewDataTypeString(), intervalEval, parser.Pos{Line: 0, Column: 0})
+	if err != nil {
+		return nil, err
+	}
+
+	interval, intervalOk := coercedInterval.(string)
+	if !intervalOk {
+		return nil, sql3.NewErrInternalf("unable to convert value")
+	}
+
+	switch strings.ToUpper(interval) {
+	case intervalYear:
+		return fmt.Sprint(date.Year()), nil
+
+	case intervalMonth:
+		return fmt.Sprintf("%d-%02d", date.Year(), date.Month()), nil
```

Comment:
I'm curious, Did you try time.Format() ? 
```
return date.Format("2006-01"), nil
return date.Format("2006-01-02"), nil
return date.Format("2006-01-02T15"), nil
.....
```

Reply:
Missed seeing that originally, converted to using it.